kroxylicious-kms-provider-hashicorp-vault-test-support/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/AbstractVaultTestKmsFacade.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.util.Set;
+
+import io.kroxylicious.kms.provider.hashicorp.vault.config.Config;
+import io.kroxylicious.kms.service.TestKmsFacade;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public abstract class AbstractVaultTestKmsFacade implements TestKmsFacade<Config, String, VaultEdek> {
+
+    protected static final String VAULT_ROOT_TOKEN = "rootToken";
+    private String kmsVaultToken;
+
+    protected AbstractVaultTestKmsFacade() {
+    }
+
+    protected abstract void startVault();
+
+    protected abstract void stopVault();
+
+    @Override
+    public final void start() {
+
+        startVault();
+
+        // enable transit engine
+        enableTransit();
+
+        // create policy
+        var policyName = "kroxylicious_encryption_filter_policy";
+        try (var policyStream = AbstractVaultTestKmsFacade.class.getResourceAsStream("kroxylicious_encryption_filter_policy.hcl")) {
+            createPolicy(policyName, policyStream);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException("Failed to create policy", e);
+        }
+        // create least-privilege vault token that KMS will use
+        kmsVaultToken = createOrphanToken("kroxylicious_encryption_filter", true, Set.of(policyName));
+    }
+
+    protected abstract void enableTransit();
+
+    protected abstract void createPolicy(String policyName, InputStream policyStream);
+
+    protected abstract String createOrphanToken(String description, boolean noDefaultPolicy, Set<String> policies);
```

Comment:
https://developer.hashicorp.com/vault/docs/concepts/tokens#token-hierarchies-and-orphan-tokens

> Is it important to the test that it's an orphan rather than a child?

No. From an Envelope Encryption point of view, orpan/no-orphan - no functional difference.  The filter neither knows nor cares.

> Or is it some best practise that you wouldn't have a root token as parent?

Disclaimer: I don't have much Vault experience so I am triangulating from other domains.

Yes.  In general you would not want the lifetime of a service account to be bound to the lifecycle of entity that creates it.

Alice's role is an  Identity Maintainer.  Her role is to create "service accounts" for applications.  Her account is suitably privileged account to let her do her work.  She almost certainly doesn't use root - least privilege/audit... etc.   Alice creates the "service account"  for Envelope Encryption.   What if she leaves? If she doesn't use `orphan`, the service account will die as her account is expired.

So, it seems to me that suggesting the `-orphan` flag was safer bet.

Of course, local practice of an organisation might mean the token should not be an orphan.   Maybe the organisation has an 'service account' token which is the parent of all the application service accounts... Who knows?

I'll add a note to the code/docs making clear that the use of orphan isn't mandatory.